### PR TITLE
feat(store-encryption): Derive Eq for some structs

### PR DESCRIPTION
**Pull request by poljar**

### CODE REVIEW

**Review by jplatte — APPROVED** [review]

File: crates/matrix-sdk-base/src/rooms/normal.rs [hunk 1]
```diff
@@ -177,7 +177,13 @@ impl Room {
...
     /// Get the avatar url of this room.
     pub fn avatar_url(&self) -> Option<OwnedMxcUri> {
-        self.inner.read().unwrap().base_info.avatar.as_ref()?.as_original()?.content.url.clone()
+        self.inner
+            .read()
+            .unwrap()
+            .base_info
+            .avatar
+            .as_ref()
+            .and_then(|e| e.as_original().and_then(|e| e.content.url.clone()))
```
[the remark this answers is not in the document]
Comment (poljar) [reply crates/matrix-sdk-base/src/rooms/normal.rs:186]: Yeah, wasn't really sure either, it was this one: https://rust-lang.github.io/rust-clippy/master/index.html#significant_drop_in_scrutinee.

[the remark this answers is not in the document]
Comment (jplatte) [reply crates/matrix-sdk-base/src/rooms/normal.rs:186]: I would say let's silence it, and maybe create a bug report in Clippy. It doesn't seem to be relevant here.

**Comment by jplatte** [comment]
Once again codecov is blocking the PR ^^°

Should we just make it not required for now?

**Comment by poljar** [comment]
> Once again codecov is blocking the PR ^^°
> 
> Should we just make it not required for now?

I vote yes.

**Comment by gnunicorn** [comment]
how is codecov blocking the PR?

**Comment by jplatte** [comment]
Screenshot 2022-05-23 at 12-54-07 feat(store-encryption) Derive Eq for some structs by poljar · Pull Request #704 · matrix-org_matrix-rust-sdk

Although it no longer shows the "Required" badge (did somebody just change required-for-merge rules already just now?)

**Comment by gnunicorn** [comment]
I had changed that last week after switching off the patch-level-coverage reports. Is this against latest master? Because patch shouldn't even show up anymore...

**Comment by poljar** [comment]
I don't think we wanted to merge this as is, considering the above discussion.

**Pull request merged**